Alignment/OfflineValidation/plugins/JetHTAnalyzer.cc

```diff
@@ -0,0 +1,360 @@
+  h_probePhi = outfile_->make<TH1F>("h_probePhi","#phi of probe track;track #phi (rad);tracks",100,-3.15,3.15);  
+  
+  h_probeDxy    = outfile_->make<TH1F>("h_probeDxy","d_{xy}(PV) of the probe track;track d_{xy}(PV);tracks",200,-100,100 );
+  h_probeDz     = outfile_->make<TH1F>("h_probeDz","d_{z}(PV) of the probe track;track d_{z}(PV);tracks",200,-100,100 );    
+  h_probeDxyErr = outfile_->make<TH1F>("h_probeDxyErr","error on d_{xy}(PV) of the probe track;track error on d_{xy}(PV);tracks",100,0.,100 );	   
+  h_probeDzErr  = outfile_->make<TH1F>("h_probeDzErr","error on d_{z}(PV)  of the probe track;track error on d_{z}(PV);tracks",100,0.,100 );	  
+
+  mon.addHistogram( new TH1F( "nvtx",";Vertices;Events",50,0,50) );
+  mon.addHistogram( new TH1F( "dxy",";d_{xy};tracks",100,-100,100) );
+  mon.addHistogram( new TH1F( "dz",";d_{z};tracks",100,-100,100) );
+  mon.addHistogram( new TH1F( "dxyerr",";d_{xy} error;tracks",100,0.,200) );
+  mon.addHistogram( new TH1F( "dzerr",";d_{z} error;tracks",100,0.,200) );
+  mon.addHistogram( new TProfile( "dxyErrVsPt",";track p_{T};d_{xy} error",100,0.,200,0.,100.));
+  mon.addHistogram( new TProfile( "dzErrVsPt" ,";track p_{T};d_{z} error" ,100,0.,200,0.,100.));
+  mon.addHistogram( new TProfile( "dxyErrVsPhi",";track #varphi;d_{xy} error",100,-TMath::Pi(),TMath::Pi(),0.,100.));
+  mon.addHistogram( new TProfile( "dzErrVsPhi" ,";track #varphi;d_{z} error" ,100,-TMath::Pi(),TMath::Pi(),0.,100.));
+  mon.addHistogram( new TProfile( "dxyErrVsEta",";track #eta;d_{xy} error",100,-2.5,2.5,0.,100.));
+  mon.addHistogram( new TProfile( "dzErrVsEta" ,";track #eta;d_{z} error" ,100,-2.5,2.5,0.,100.));
+    
+  // Variable size histogram depending on the given pT bin borders
+  int nBins = profilePtBorders_.size();
+  mon.addHistogram( new TProfile( "dxyErrVsPtWide", ";track p_{T} wide bin;d_{xy} error", nBins, -0.5, nBins-0.5, 0.0, 100.0));
+  mon.addHistogram( new TProfile( "dzErrVsPtWide", ";track p_{T} wide bin;d_{z} error", nBins, -0.5, nBins-0.5, 0.0, 100.0));
+
+}
+
+// ------------ method called once each job just after ending the event loop  ------------
+void
+JetHTAnalyzer::endJob()
+{
+  mon.Write();
+}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void
+JetHTAnalyzer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+  desc.setUnknown();
+  descriptions.addDefault(desc);
+
+  //Specify that only 'tracks' is allowed
+  //To use, remove the default given above and uncomment below
+  //ParameterSetDescription desc;
+  //desc.addUntracked<edm::InputTag>("tracks","ctfWithMaterialTracks");
+  //descriptions.addDefault(desc);
```

Comment:
this default text could be cleaned up

Reply:
That explains why text bellow is commented out and when it should be un-commented. Therefore I would leave it.

Reply:
actually a real `fillDescriptions` method should be provided and the boiler plate code removed.

Reply:
fillDescriptions was added